google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/standard/AppEngineStandardWebIntegration.java

```diff
@@ -42,21 +44,35 @@
 import com.intellij.remoteServer.impl.configuration.deployment.DeployToServerConfigurationType;
 import com.intellij.remoteServer.impl.configuration.deployment.DeployToServerConfigurationTypesRegistrar;
 import com.intellij.remoteServer.impl.configuration.deployment.DeployToServerRunConfiguration;
+import com.intellij.util.ArrayUtil;
 import com.intellij.util.containers.ContainerUtil;
+import java.io.IOException;
 import java.util.List;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 /** @author nik */
 public abstract class AppEngineStandardWebIntegration {
+  private static final Logger LOG = Logger.getInstance(AppEngineStandardWebIntegration.class);
 
   public static AppEngineStandardWebIntegration getInstance() {
     return ServiceManager.getService(AppEngineStandardWebIntegration.class);
   }
 
   @Nullable
-  public abstract VirtualFile suggestParentDirectoryForAppEngineWebXml(
-      @NotNull Module module, @NotNull ModifiableRootModel rootModel);
+  public VirtualFile suggestParentDirectoryForAppEngineWebXml(
```

Comment:
can you explain a bit (just so I understand) the idea behind moving the implementation to the base integration class? Before, the Ultimate implementation didn't fall back to this strategy (by calling super), but in the PR it is. Just trying to wrap my head around it.

Reply:
This method creates the appengine-web.xml directory from a static path from the root of the project. 

In IC - this is only what it did using the WEB-INF path 
In IU - if there was no web.xml, this method returned null (hence why the appengine-web.xml was not getting generated in #1903). With the PR, if web.xml doesn't exist and there are no web resource directories, then it creates the parent of the appengine-web.xml in a from a static path too.